### ISSUE TITLE
[BUG] Can't install Harvester cloud provider in Rancher v2.7-head guest cluster configuration page  

### ISSUE THREAD

**Issue by TachunLin** [harvester/harvester#3506]
**Describe the bug**

On rancher v2.7-head, in the RKE2 cluster provisioning page, we can't select Harvester cloud provider for the given Kubernetes version of `v1.25.6`, `v1.24.10` and `v1.23.6` 
![image](https://user-images.githubusercontent.com/29251855/220071850-145f46fa-7f84-484a-96a6-10b09de4cff9.png)

**To Reproduce**
1. Import Harvester to Rancher 
2. Create cloud credential 
3. Provision RKE2 cluster 
4. Check the Cloud provider setting drop down list

**Expected behavior**

Should be able to select Harvester cloud provider for the given Kubernetes version 

**Support bundle**

**Environment**
 - Harvester ISO version: v1.1.2-rc1
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): local kvm machine
 - Rancher version: v2.7-head (ab561bf)
 - Node driver version: v0.6.1

**Additional context**

**Comment by n313893254** [comment]
Same as an issue [3389](https://github.com/harvester/harvester/issues/3389). The Harvester cloud provider can not select [issue](https://github.com/harvester/harvester/issues/3404) is caused by the version setting check: `/v1/harvester/harvesterhci.io.settings/harvester-csi-ccm-versions`. 

~The workaround is to update the setting default value or value to `{"harvester-cloud-provider":">=0.0.1","harvester-csi-provider":">=0.0.1"}`~

The workaround is to change the Harvester server-version setting to v1.1.2.

![image](https://user-images.githubusercontent.com/18737885/220052565-8546c0e1-127b-426b-adf0-6742ebe6b98b.png)

**Comment by guangbochen** [comment]
@n313893254 doesn't the upstream PR https://github.com/rancher/dashboard/pull/8081 already address the issue?

**Comment by n313893254** [comment]
@guangbochen This PR is valid for official versions like v1.1.2, but no judgment is made for RC versions that have a suffix. The workaround is to change the Harvester server-version setting to v1.1.2.

**Comment by irishgordo** [comment]
@n313893254 thanks for the workaround :smile: 
This looks good, I'll go ahead and close this out.

Validated on:
- Harvester Version: v1.1.2-rc1
- Rancher Version v2.7-head / v2.7-eb96417fb0fc75bbb5c4dde95aed949d6585239d-head

With the workaround of:
- Navigate to Harvester "admin" preferences
- Check boxes:
    - "Enable View in API"
    - "Show system Namespaces managed by Rancher"
    - "Enable Extension Developer features"
- Navigate to Harvester -> Advanced -> Settings -> `server-version` -> `View in API` -> `Edit` (removing "-rc1") -> `Send Request`, validate 200

Then import Harvester to Rancher.
Validate current KDM Metadata for v2.7-head has latest drivers:
```
╭─mike at suse-workstation-team-harvester in ~
╰─○ curl https://releases.rancher.com/kontainer-driver-metadata/dev-v2.7/data.json | jq '.rke2.releases[] | select( .charts["harvester-csi-driver"]["version"] == "0.1.1500" ) | {"rke2-releases-rke2-version": .version, "rke2-releases-rke2-provided-harvester-csi-driver": .charts["harvester-csi-driver"]}'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 5311k  100 5311k    0     0  17.6M      0 --:--:-- --:--:-- --:--:-- 17.6M
{
  "rke2-releases-rke2-version": "v1.23.16+rke2r1",
  "rke2-releases-rke2-provided-harvester-csi-driver": {
    "repo": "rancher-rke2-charts",
    "version": "0.1.1500"
  }
}
{
  "rke2-releases-rke2-version": "v1.24.10+rke2r1",
  "rke2-releases-rke2-provided-harvester-csi-driver": {
    "repo": "rancher-rke2-charts",
    "version": "0.1.1500"
  }
}
{
  "rke2-releases-rke2-version": "v1.25.6+rke2r1",
  "rke2-releases-rke2-provided-harvester-csi-driver": {
    "repo": "rancher-rke2-charts",
    "version": "0.1.1500"
  }
}

```

Validate Harvester is not greyed out and can be installed:
![Screenshot from 2023-02-22 13-42-01](https://user-images.githubusercontent.com/5370752/220766846-750ceb60-a863-4b7f-9776-66e477c44e1f.png)
![Screenshot from 2023-02-22 13-41-29](https://user-images.githubusercontent.com/5370752/220766851-eabc3161-91f7-48de-841b-44ae2d31085d.png)
![Screenshot from 2023-02-22 13-41-07](https://user-images.githubusercontent.com/5370752/220766852-87c33973-4214-4e29-89dd-d83d139e3a28.png)
![Screenshot from 2023-02-22 13-40-44](https://user-images.githubusercontent.com/5370752/220766854-c3390cc0-7594-4ce0-83cd-1657fde8a200.png)
![Screenshot from 2023-02-22 13-38-46](https://user-images.githubusercontent.com/5370752/220766855-5264c61a-b820-456b-a4a1-4298438f007f.png)
![Screenshot from 2023-02-22 13-38-22](https://user-images.githubusercontent.com/5370752/220766857-e3fb0b73-34a2-4cb5-b677-018b22a1d503.png)
![Screenshot from 2023-02-22 13-37-54](https://user-images.githubusercontent.com/5370752/220766859-90e1ed7c-53eb-441c-bce2-f35abc6d574a.png)
![Screenshot from 2023-02-22 13-37-29](https://user-images.githubusercontent.com/5370752/220766860-520e2233-fa05-4ff8-ae56-155c649b805b.png)
![Screenshot from 2023-02-22 13-54-39](https://user-images.githubusercontent.com/5370752/220768312-73048e05-529e-47b9-96a2-2d3862f80f58.png)

**Comment by WuJun2016** [comment]
@irishgordo   This problem has been fixed in the rancher 2.7-head, and it does not need to use the workaround method to test.     Can you help me use the latest rancher to test it?  thank you

**Comment by WuJun2016** [comment]
I'll reopen this issue, thank you

**Comment by irishgordo** [comment]
@WuJun2016 - I can help validate with Rancher v2.7-head :smile: and not implementing the previous workaround - I'll work on validating this now

**Comment by irishgordo** [comment]
@WuJun2016 
This looks good on Rancher v2.7-head ( v2.7-7c60dc0856e19c278d520946d4d2e7ea2b86d70f-head ) with Harvester v1.1.2-rc1.
No workaround was implemented, so the `server-version` was left un-touched :smile: .

![Screenshot from 2023-02-22 18-18-51](https://user-images.githubusercontent.com/5370752/220807566-24cc3d24-dd17-42b1-9051-391b010879ca.png)
![Screenshot from 2023-02-22 18-10-27](https://user-images.githubusercontent.com/5370752/220807569-a18d2bf0-1b1a-4f15-b14f-0af75d8eaf07.png)
![Screenshot from 2023-02-22 18-10-04](https://user-images.githubusercontent.com/5370752/220807572-268de334-edc0-4127-ace4-8c57f0ac206e.png)
![Screenshot from 2023-02-22 18-09-39](https://user-images.githubusercontent.com/5370752/220807574-82593ec2-a300-4d3f-86aa-9b5f2e38c9e0.png)

Should I go ahead and close this out :smile: ?

**Comment by WuJun2016** [comment]
@irishgordo   Yes, you can close the issue. Thank you for your verification.

**Comment by irishgordo** [comment]
You're welcome!  Glad I could help! :smile: Thanks again for pre-release comparison logic fix @WuJun2016 !

**Comment by guangbochen** [comment]
This will be fixed in the upcoming Rancher v2.7.2, as a temp workaround, you can modify the related setting in the Harvester via:
```
$ kubectl edit settings.harvesterhci.io harvester-csi-ccm-versions

# and patch the default value as below
default: '{"harvester-cloud-provider":">=0.0.1 <0.2.0","harvester-csi-provider":">=0.0.1 <0.2.0"}'
```
